internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -811,12 +818,20 @@ func expandProcessArgs(d *schema.ResourceData, p map[string]any) (admin20240530.
 		}
 	}
 
-	if _, ok := d.GetOkExists("advanced_configuration.0.change_stream_options_pre_and_post_images_expire_after_seconds"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.change_stream_options_pre_and_post_images_expire_after_seconds"); ok && IsChangeStreamOptionsMinRequiredVersion(mongodbMajorVersion) {
 		res.ChangeStreamOptionsPreAndPostImagesExpireAfterSeconds = conversion.IntPtr(cast.ToInt(p["change_stream_options_pre_and_post_images_expire_after_seconds"]))
 	}
 	return res20240530, res
 }
 
+func IsChangeStreamOptionsMinRequiredVersion(input *string) bool {
+	if input == nil || *input == "" {
+		return true
+	}
+	value, _ := strconv.ParseFloat(*input, 64)
```

Comment:
I would like to see an acceptance test or at least unit tests

Reply:
Yes I see the github issue mentions `5.0.28`, this would fail to parse as a float

Reply:
This method is intended to be used for MongoDBMajorVersion. Major version is not expected to have that format. 
I will still update the method to only take into account the first digit of the version passed so it can be more generic.

Reply:
updated the method as mentioned above, also added an acceptance and unit test